### ISSUE TITLE
Follow up on #39 - publish to html

### ISSUE THREAD

**Issue by mikkelking** [opticdev/optic#244]
I am trying to follow the advice in #39 

```
   start:
     command: npm run start
     baseUrl: http://localhost:3700
   publish:
     oas: |
       widdershins {{{OAS_PATH}}} -o /tmp/api.md
       shins --inline -o ../doc/index.html /tmp/api.md
```
But this doesn't work for me. 

` ›   Error: command publish not found`

I can run these command from the command line, and I could create an npm script for them. I have tried a few combinations, including taking out the pipe after `oas:`, but nothing seems to work

**Comment by LouManglass** [comment]
Hi @mikkelking,

Thanks for the report. I'm digging into this further now. We generally recommend viewing the API specification in Optic with `api spec`, which will spin up a local dashboard. The second option down on the left-hand nav bar, an icon of a document, will show you the current documentation. Anyone with a clone of the repository (with the .optic directory committed) and Optic installed will be able to view this spec as well.

In case you need to be able to export to OAS, the generate OAS feature is available from the command line: `api generate:oas` will create an OAS file in JSON format (optionally the command takes the `--yaml` flag for YAML format) which you can then use in your tooling of choice.

Let me know if the above options work for you while I dig into the advice given in #39 and get you a full answer.

**Comment by LouManglass** [comment]
Howdy,

generating the spec from a task in the optic.yml file is a deprecated path, in favor of `api generate:oas` directly on the command line. Will this work for your use case? If so, I can update our recommendation in #39 as well to avoid future confusion.

If not, please let me know more about your particular scenario and what you'd like to see.

**Comment by LouManglass** [comment]
Updated #39 to reflect the `generate:oas` command.

**Comment by mikkelking** [comment]
I like the idea of using shins, because I want to publish it to a static web site, and I don't think optic can do that.
As for this specific problem, I can get around it by chaining together the `api generate:oas` and the shins commands to get what I want.

I have also taken the OAS file and dropped it into Insomnia Designer, which also gives me the options of being able to try the api calls from the spec.

I am quite impressed at what Optic does. We are building API's first, and then documenting them after the fact with Optic.

**Comment by LouManglass** [comment]
That's correct: Optic currently cannot publish your spec to a static web site. Our goal is to make it easy to share your documented API specification with your team (or everyone!). I'll pass your feedback along: is it fair to say that a static, portable documentation would be handy so that you can host it wherever it makes sense? Would it help if we had an option available to publish live documentation for you, so you didn't have to export the static site and put it somewhere?


Thanks for the feedback! We're eager to see how Optic works out for you and how well it works with your tooling. If you have other questions or feedback, we're happy to hear it. We'd also be happy to jump on a call if you want to point out anything in particular directly. Have a great time, and we look forward to adding more impressive features to help you focus on building your APIs.

**Comment by LouManglass** [comment]
Hi @mikkelking,

We just launched the Optic Scrips feature, which lives alongside Optic Tasks as a full-fledged, post-processing tool to address this issue. Optic Scripts are run from the command line with `optic scripts`, are defined in `optic.yml`, and automatically generate the latest OAS file which can then be manipulated by a script.

For example, we published a blog post using [widdershins and shins](https://www.useoptic.com/blog/optic-scripts) with the new Optic Scripts. Documentation is available at [Configuring Scripts](https://www.useoptic.com/docs/faqs-and-troubleshooting/scripts). Let me know if this helps keeps things running smoothly, and if we can make it work better for you.
